core/block_validator.go

```diff
@@ -155,30 +122,6 @@ func (v *BlockValidator) ValidateGasAndRoot(block *types.Block, root common.Hash
 		fmt.Fprintf(&errorBuf, "invalid gas used (remote: %d local: %d)", block.GasUsed(), usedGas)
 	}
 
-	// Validate the state root against the received state root and throw
-	// an error if they don't match.
-	if block.Header().Root != root {
```

Comment:
You should keep this line, and also the code starting from the line 168:
```
if errorBuf.Len() > 0 {
			errorBuf.WriteString("; ")
		}
		fmt.Fprintf(&errorBuf, "[pre-processed] invalid merkle root (remote: %x local: %x)", block.Header().Root, root)
```

otherwise tests won't pass